contracts/src/mixins/QuotaMixin.sol

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: GNU GPLv3
+pragma solidity ^0.8.29;
+
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/QuotaManager.sol";
+import "../libs/CoreLib.sol";
+
+struct QuotaStats {
+    uint128 tokenBalance; // uint96 might also be enough, as this would be more tokens than most project have in circulation
```

Comment:
`uint96` is enough for Uniswap v2 pools, so it is enough for pretty much everything.

That being said, you have the 4 bytes to spare, so no harm in using them. You can take from the `tokenBalance` if ever you need a few more bits.